test/c/test_substrait_c_api.cpp

```diff
@@ -341,20 +341,31 @@ TEST_CASE("Test C VirtualTable input Literal", "[substrait-api]") {
 
   auto json = GetSubstraitJSON(con,"select * from (values (1, 2),(3, 4))");
   REQUIRE(!json.empty());
-  std::cout << json << std::endl;
 
   auto result = FromSubstraitJSON(con,json);
   REQUIRE(CHECK_COLUMN(result, 0, {1, 3}));
   REQUIRE(CHECK_COLUMN(result, 1, {2, 4}));
 }
 
+TEST_CASE("Test C VirtualTable with bad input Literal", "[substrait-api]") {
+	DuckDB db(nullptr);
+	Connection con(db);
+
+	auto json_plan = R"({"relations":[{"root":{"input":{"project":{"input":{"project":{"input":{"read":{"virtualTable":{"expressions":[{"fields":[{"literal":{"decimal":{"value":"pAYAAAAAAAAAAAAAAAAAAA==","precision":11,"scale":1}}},{"literal":{"decimal":{"value":"TgcAAAAAAAAAAAAAAAAAAA==","precision":4,"scale":1}}}]},{"fields":[{"literal":{"decimal":{"value":"+AcAAAAAAAAAAAAAAAAAAA==","precision":11,"scale":1}}},{"literal":{"decimal":{"value":"AQ==","precision":4,"scale":0}}}]}]}}},"expressions":[{"selection":{"directReference":{"structField":{}},"rootReference":{}}},{"selection":{"directReference":{"structField":{"field":1}},"rootReference":{}}}]}},"expressions":[{"selection":{"directReference":{"structField":{}},"rootReference":{}}},{"selection":{"directReference":{"structField":{"field":1}},"rootReference":{}}}]}},"names":["col0","col1"]}}],"version":{"minorNumber":53,"producer":"DuckDB"}})";
```

Comment:
Since you just want to test a literal table a single virtual table without any projects should be enough.